### ISSUE TITLE
[AOT] Link failed on Apple silicon for LLVM-15 or higher

### ISSUE THREAD

**Issue by q82419** [WasmEdge/WasmEdge#2435]
### Current State

When using LLVM-15 or higher on Apple silicon platforms, linking will fail when using `-DWASMEDGE_LINK_LLVM_STATIC=ON` flag.

### Expected

Link success.

## Environment

- Hardware Architecture: aarch64 (M2 max)
- Operating system: MacOS 13.2.1

## Steps to Reproduce

```
brew install llvm # or install llvm@15
cmake -Bbuild -GNinja -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_LINK_LLVM_STATIC=ON -DWASMEDGE_BUILD_PACKAGE="TGZ" .
cmake --build build --target package
```

**Comment by hydai** [comment]
@q82419 
What's the error log?

**Comment by q82419** [comment]
When linking `libwasmedge.dylib`:

```
undef: _ZSTD_compress
undef: _ZSTD_compressBound
undef: _ZSTD_decompress
undef: _ZSTD_getErrorName
undef: _ZSTD_isError
Undefined symbols for architecture arm64:
  "_ZSTD_compress", referenced from:
      llvm::compression::zstd::compress(llvm::ArrayRef<unsigned char>, llvm::SmallVectorImpl<unsigned char>&, int) in libLLVMSupport.a(Compression.cpp.o)
  "_ZSTD_compressBound", referenced from:
      llvm::compression::zstd::compress(llvm::ArrayRef<unsigned char>, llvm::SmallVectorImpl<unsigned char>&, int) in libLLVMSupport.a(Compression.cpp.o)
  "_ZSTD_decompress", referenced from:
      llvm::compression::zstd::uncompress(llvm::ArrayRef<unsigned char>, unsigned char*, unsigned long&) in libLLVMSupport.a(Compression.cpp.o)
  "_ZSTD_getErrorName", referenced from:
      llvm::compression::zstd::uncompress(llvm::ArrayRef<unsigned char>, unsigned char*, unsigned long&) in libLLVMSupport.a(Compression.cpp.o)
  "_ZSTD_isError", referenced from:
      llvm::compression::zstd::compress(llvm::ArrayRef<unsigned char>, llvm::SmallVectorImpl<unsigned char>&, int) in libLLVMSupport.a(Compression.cpp.o)
      llvm::compression::zstd::uncompress(llvm::ArrayRef<unsigned char>, unsigned char*, unsigned long&) in libLLVMSupport.a(Compression.cpp.o)
ld: symbol(s) not found for architecture arm64
```

**Comment by q82419** [comment]
Seems like fixed by #2447
